src/Features/CSharp/Portable/InitializeParameter/InitializeParameterHelpers.cs

```diff
@@ -29,7 +32,7 @@ public static IBlockOperation GetBlockOperation(SyntaxNode functionDeclaration,
 
             // body might be an expression in a lambda - in that case we wouldn't get a block operation out of that
             return functionDeclaration is AnonymousFunctionExpressionSyntax
-                ? ((IAnonymousFunctionOperation)semanticModel.GetOperation(functionDeclaration, cancellationToken)).Body
+                ? ((IAnonymousFunctionOperation)operation).Body
                 : (IBlockOperation)semanticModel.GetOperation(bodyOpt, cancellationToken);
```

Comment:
I am not sure we should directly cast to `IBlockOperation` based on syntax here.